### PR TITLE
OCPBUGS-13147: Add golangci-lint linters

### PR DESCRIPTION
**Pull request by danielmellado**
This commit adds a new makefile target `golangci-lint` which allows the
usage of it for linting.

It also adds a default configuration including
static checking and fixes its staticcheck violations.
In order to do so, it also updates go mod vendoring.

Signed-off-by: Daniel Mellado <dmellado@redhat.com>

### CODE REVIEW

**Review by marioferh — COMMENTED** [review]
/lgtm

File: .golangci.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+run:
```
[the remark this answers is not in the document]
Comment (jan--f) [reply .golangci.yaml:1]: The timeout is set via cli argument in the Makefile call below. 
But I see `modules-download-mode: vendor` is also passed via cli arg.
Having on place makes certainly sense. I'd slightly lean toward specifying everything in the config file, but no strong opinion. Except to set an individual config in one place only.
@danielmellado wdyt?

**Comment by jan--f** [comment]
/label docs-approved
/label px-approved
/label qe-approved
Adding labels for an internal test addition. 
Largely this looks good, we should find consensus and consistency on cli args vs config file.

File: .golangci.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (danielmellado) [reply .golangci.yaml:1]: Yeah, I put the timeout on the `golangci-lint' call directly on the makefile as I thought it'd be more visible there, but not a strong opinion on there. I'd keep it there, but I'll add a comment stating the why if that works for you.

@vimalk78 it's on the Makefile, see -> ```$(GOLANGCI_LINT_BIN) run --verbose --print-resources-usage --modules-download-mode=vendor --timeout=5m0s```

[the remark this answers is not in the document]
Comment (danielmellado) [reply .golangci.yaml:1]: In the end, after considering it, removed both download mode and timeout from the cli invoke and put that in the config. Should be good now ;)

[the remark this answers is not in the document]
Comment (jan--f) [reply .golangci.yaml:1]: @danielmellado you could add `-c .golangci.yaml` to the cli call. That should make it fairly explicit?

File: Makefile [hunk 2]
```diff
@@ -228,3 +234,4 @@ $(TOOLING): $(BIN_DIR)
 	@echo Installing tools from hack/tools.go
 	@cd hack/tools && go list -mod=mod -tags tools -f '{{ range .Imports }}{{ printf "%s\n" .}}{{end}}' ./ | xargs -tI % go build -mod=mod -o $(BIN_DIR) %
 	@GOBIN=$(BIN_DIR) go install $(GO_PKG)/hack/docgen
+	curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b $(BIN_DIR) $(GOLANGCI_LINT_VERSION)
```
Comment (simonpasquier) [inline Makefile:237]: from past experience, it'd better to pin a know version otherwise new checks could mean that suddenly our CI is blocked.

Comment (danielmellado) [reply Makefile:237]: It actually is, see `GOLANGCI_LINT_VERSION=v1.52.2` env var over there ;)

Comment (simonpasquier) [reply Makefile:237]: ah my bad!

**Pull request merged**